lib/components/PolicyMap/PolicyMap.js

```diff
@@ -0,0 +1,223 @@
+    numeral(sorted[Math.floor(0.5 * sorted.length)].value + 1).format('0a'),
+    numeral(sorted[Math.floor(0.75 * sorted.length)].value).format('0a'),
+    numeral(sorted[Math.floor(0.75 * sorted.length)].value + 1).format('0a'),
+    numeral(sorted[sorted.length - 1].value).format('0a'),
+  ];
+
+  sorted = sorted.map((dataPoint, i) => {
+    const colorIndex = Math.floor((i / sorted.length) * COLOR_SCALE.length);
+
+    // Color the country according to quartile if none is selected
+    if (!focusQuartile) {
+      return { ...dataPoint, color: COLOR_SCALE[colorIndex] };
+    }
+
+    // Is a quartile is selected, only color the country if it falls in the quartile range
+    if (focusQuartile === colorIndex + 1) {
+      return { ...dataPoint, color: COLOR_SCALE[colorIndex] };
+    }
+    return {};
+  });
+
+  const mapContent = ({ geographies }) => {
+    const countries = [];
+    const labels = [];
+    geographies.forEach((geo) => {
+      const dataPoint = sorted.find((point) => point.id === geo.properties.ISO_A2);
+      countries.push(
+        <Geography
+          key={geo.rsmKey}
+          geography={geo}
+          onClick={() => dataPoint && onClick(dataPoint)}
+          fill={(dataPoint && dataPoint.color) || INACTIVE_COLOR}
+          stroke={WHITE}
+          strokeWidth="1.2px"
+          vectorEffect="non-scaling-stroke"
+          onMouseEnter={() => {
+            if (dataPoint && dataPoint.value) {
+              setTooltip({
+                name: geo.properties.NAME,
+                value: numeral(dataPoint.value).format('0a'),
+              });
+            }
+          }}
+          onMouseLeave={() => {
+            setTooltip('');
+          }}
+          style={{
+            hover: {
+              opacity: 0.7,
+              // fill: dataPoint && dataPoint.value && HIGHLIGHT_COLOR,
+              outline: 'none',
+            },
+          }}
+        />,
+      );
+    });
+    return (
+      <>
+        {countries}
+        {labels}
+      </>
+    );
+  };
+
+  return (
+    <div className={styles.container}>
+      <ReactTooltip backgroundColor={WHITE} borderColor={BORDER_COLOR} border textColor="#000000">
+        {tooltip && (
+          <p className={styles.tooltip}>
+            <strong>{tooltip.name}</strong>
+            <br />
+            <span>
+              {labelText}: {tooltip.value}
+            </span>
+          </p>
+        )}
+      </ReactTooltip>
+      <ComposableMap
+        data-tip=""
+        projection="geoMercator"
+        projectionConfig={{
+          rotate: [-10, 0, 0],
+          scale: 125,
+        }}
+      >
+        <ZoomableGroup zoom={1}>
+          <Geographies geography={geoData}>{mapContent}</Geographies>
+        </ZoomableGroup>
+      </ComposableMap>
+      <legend className={styles.legendContainer}>
+        <ul className={styles.legendList}>
+          <li className={styles.legendListItem}>
```

Comment:
we can simplify things a little bit with these changes:
```jsx
const format = (value) => numeral(value).format('0a');
const pair = (result, value, index, array) => {
  if (index % 2 === 0) {
    result.push(array.slice(index, index + 2));
  }

  return result;
};

// Breakpoints used for legend
const breakpoints = [
  1,
  sorted[Math.floor(0.25 * sorted.length)].value,
  sorted[Math.floor(0.25 * sorted.length)].value + 1,
  sorted[Math.floor(0.5 * sorted.length)].value,
  sorted[Math.floor(0.5 * sorted.length)].value + 1,
  sorted[Math.floor(0.75 * sorted.length)].value,
  sorted[Math.floor(0.75 * sorted.length)].value + 1,
  sorted[sorted.length - 1].value,
]
  .map(format)
  .reduce(pair, []);
```

and 

```jsx
{breakpoints.map((breakpointsPair, index) => (
  // eslint-disable-next-line react/no-array-index-key
  <li key={index} className={styles.legendListItem}>
    <button
      className={cn(
        styles.legendListButton,
        focusQuartile && focusQuartile !== index + 1 && styles.legendListDisabled,
      )}
      onClick={() => toggleQuartile(index + 1)}
      type="button"
    >
      <div className={cn(styles.legendBox, styles[`quartile${index + 1}`])} />
      <span>{breakpointsPair.join(' - ')}</span>
    </button>
  </li>
))}
```

Reply:
![](https://thumbs.gfycat.com/EachFabulousAsianpiedstarling-size_restricted.gif)
good idea, done